src/Sarif/Baseline/ResultMatching/SarifLogMatcher.cs

```diff
@@ -279,20 +279,12 @@ private ReportingDescriptor GetRuleFromResources(Result result, IDictionary<stri
             var graphs = new Dictionary<string, Graph>();
             var ruleData = new Dictionary<string, ReportingDescriptor>();
             var invocations = new List<Invocation>();
-            var messageData = new Dictionary<string, string>();
+
+            // TODO tool message strings are not currently handled
+            // https://github.com/Microsoft/sarif-sdk/issues/1286
 
             foreach (Run currentRun in currentRuns)
             {
-                if (currentRun.Tool.Driver.GlobalMessageStrings != null)
-                {
-                    IDictionary<string, string> converted = currentRun.Tool.Driver.GlobalMessageStrings;
-                    if (converted == null)
-                    {
-                        throw new ArgumentException("Message Strings did not deserialize properly into a dictionary mapping strings to strings.");
-                    }
-                    MergeDictionaryInto(messageData, converted, StringComparer.InvariantCulture);
```

Comment:
>MergeDictionaryInto [](start = 20, length = 19)

Just curious why this didn't work when the values in `GlobalMessageStrings` changed from `string` to `MultiformatMessageString`. `MergeDictionaryInto` is generic and uses a value comparer (which we have), so naively I'd've expected this to just work.